### ISSUE TITLE
[BUG] 404 when saving settings

### ISSUE THREAD

**Issue by NerdRelaxo** [FreshRSS/FreshRSS#4941]
**Describe the bug**
I made a fresh install on my raspberrypi. Everything went fine, I imported the feeds from my old installation which will be shut down. Then I wanted to change e.g. the read settings. When I click on the save-button, I get an 404 error.

**To Reproduce**
1. Go to 'Configuration/read'
2. Click on 'change any setting and click on save'
3. See error

**Expected behavior**
Saving the settings.

**Screenshots**
<img width="410" alt="image" src="https://user-images.githubusercontent.com/120359401/206973198-77059638-e554-47da-9e27-2dc760c61be3.png">

**Environment information (please complete the following information):**
- Device: raspberrypi 4B/8GB
- OS: Linux version 5.15.76-v8+ (dom@buildbot) (aarch64-linux-gnu-gcc-8 (Ubuntu/Linaro 8.4.0-3ubuntu1) 8.4.0, GNU ld (GNU Binutils for Ubuntu) 2.34) #1597 SMP PREEMPT
- Browser: Safari
- FreshRSS version: 1.20.1
- Database version: litesql
- PHP version:  PHP 7.4
- Installation type: raspberry

In the browser console I see this:

<img width="1001" alt="image" src="https://user-images.githubusercontent.com/120359401/206975175-24fd7e49-ddde-4306-ae21-b28d6813235e.png">

I think the problem ist the 302 as answer to the POST request. This makes the browser to follow and make a GET request instead. If the redirect is needed, it should be a 307.

It's my first report here on GitHub. So please be kind if I missed something. I didn't find anyone with this issue here with the search function.

**Comment by NerdRelaxo** [comment]
I just noticed that I'm getting that 404 also when I press "mark all articles as read" on the bottom of the feed.

**Comment by math-GH** [comment]
Any log messages?

How did you export/import the feeds?

**Comment by Alkarex** [comment]
Hello,
Please check in your `./data/config.php` file the line `base_url`, which might be wrong in your case.

Maybe same than https://github.com/FreshRSS/FreshRSS/discussions/4677 (in French)

What Web server are you using (if you are not using Docker), and do you have any proxy in front?

**Comment by NerdRelaxo** [comment]
@Alkarex I'm using Apache, the config you mentioned should be okay as far as I can see from the other issue:

'base_url' => 'http://rss.nerdcave.local:8080',

I think the path is also correct, as I see in the Headers. The site sends a POST to /I/ which is below the vHost path /var/www/FreshRSS/p. I wonder where the redirect comes from. The redirect won't be a problem if it would be a 307. 

@math-GH /tmp/FreshRSS.log has only one line:

/bin/sh: 1: www-data: not found

Where do I find another log? The access.log from the Apache only gives me what I already see in the Headers, a POST resulting in 302 and then a GET resulting in the 404.

**Comment by NerdRelaxo** [comment]
Nervermind, I just found out that the error only occurs in Safari 16.1 in Firefox and on my iPad Safari it works. oO

**Comment by mappu** [comment]
> Please check in your `./data/config.php` file the line `base_url`, which might be wrong in your case.

Hi, thanks for FreshRSS. I just want to confirm I experienced the same (or similar?) issue after upgrading my Docker container from 1.19.0-alpine to 1.20.2-alpine.

In the config.php I had `base_url` set to `https://mydomain/p` . 

Everything was working in 1.19.0 (and all past versions i used going back to circa 1.14). But after upgrading to 1.20.2-alpine I experienced the following:

- Visiting the mydomain redirects to mydomain/p/i/... which 404s
- Manually visiting mydomain/i/ works, can log in, read articles
- Saving the settings redirects to mydomain/p/i/... which 404s
- Clicking the "Mark all as read" button also redirects to mydomain/p/i/ which 404s. However manually sending the browser back to mydomain/i/ works and the articles were correctly marked as read.

This was a browser-agnostic problem (Chromium on PC, Firefox on mobile, ...)

After your comment above I changed the config.php `base_url` to just `https://mydomain/`

This seems to have fully resolved everything. The `/p/` is not present in the URLs any more, so there was a structural URL format change, but all links and buttons are now working normally again.
